src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -0,0 +1,717 @@
+        UnsignedMultiply,
+        UnsignedDivide,
+        UnsignedRemainder,
+        UnsignedLeftShift,
+        UnsignedRightShift,
+        UnsignedAnd,
+        UnsignedOr,
+        UnsignedExclusiveOr,
+
+        FloatingAdd,
+        FloatingSubtract,
+        FloatingMultiply,
+        FloatingDivide,
+        FloatingRemainder,
+        FloatingPower,
+
+        DecimalAdd,
+        DecimalSubtract,
+        DecimalMultiply,
+        DecimalDivide,
+
+        BooleanAnd,
+        BooleanOr,
+        BooleanExclusiveOr,
+        BooleanConditionalAnd,
+        BooleanConditionalOr,
+
+        EnumAdd,
+        EnumSubtract,
+        EnumAnd,
+        EnumOr,
+        EnumExclusiveOr,
+
+        PointerIntegerAdd,
+        IntegerPointerAdd,
+        PointerIntegerSubtract,
+        PointerSubtract,
+
+        DynamicAdd,
+        DynamicSubtract,
+        DynamicMultiply,
+        DynamicDivide,
+        DynamicRemainder,
+        DynamicLeftShift,
+        DynamicRightShift,
+        DynamicAnd,
+        DynamicOr,
+        DynamicExclusiveOr,
+
+        ObjectAdd,
+        ObjectSubtract,
+        ObjectMultiply,
+        ObjectDivide,
+        ObjectPower,
+        ObjectIntegerDivide,
+        ObjectRemainder,
+        ObjectLeftShift,
+        ObjectRightShift,
+        ObjectAnd,
+        ObjectOr,
+        ObjectExclusiveOr,
+        ObjectConditionalAnd,
+        ObjectConditionalOr,
+        ObjectConcatenation,
+
+        StringConcatenation,
+
+        // Relational operations.
+
+        OperatorEqual,
+        OperatorNotEqual,
+        OperatorLess,
+        OperatorLessEqual,
+        OperatorGreaterEqual,
+        OperatorGreater,
+
+        IntegerEqual,
+        IntegerNotEqual,
+        IntegerLess,
+        IntegerLessEqual,
+        IntegerGreaterEqual,
+        IntegerGreater,
+        UnsignedLess,
+        UnsignedLessEqual,
+        UnsignedGreaterEqual,
+        UnsignedGreater,
+
+        FloatingEqual,
+        FloatingNotEqual,
+        FloatingLess,
+        FloatingLessEqual,
+        FloatingGreaterEqual,
+        FloatingGreater,
+
+        DecimalEqual,
+        DecimalNotEqual,
+        DecimalLess,
+        DecimalLessEqual,
+        DecimalGreaterEqual,
+        DecimalGreater,
+
+        BooleanEqual,
+        BooleanNotEqual,
+
+        StringEqual,
+        StringNotEqual,
+        StringLike,
+
+        DelegateEqual,
+        DelegateNotEqual,
+
+        NullableEqual,
+        NullableNotEqual,
+
+        ObjectLess,
+        ObjectLessEqual,
+        ObjectEqual,
+        ObjectNotEqual,
+        ObjectVBEqual,
+        ObjectVBNotEqual,
+        ObjectLike,
+        ObjectGreaterEqual,
+        ObjectGreater,
+
+        EnumEqual,
+        EnumNotEqual,
+        EnumLess,
+        EnumLessEqual,
+        EnumGreaterEqual,
+        EnumGreater,
+
+        PointerEqual,
+        PointerNotEqual,
+        PointerLess,
+        PointerLessEqual,
+        PointerGreaterEqual,
```

Comment:
They have been reviewed in the sense of being looked at by a group of people focused on other things. The names in the bound trees are of the form "GreaterThanOrEqual". I'm happy to adopt this convention.

I'm hoping to avoid flags enums, and we can have all desired helpers.